### PR TITLE
Remove GOV.UK Frontend Toolkit [part 5]

### PR DESCRIPTION
**Pull request by klssmith**
This is part 5 of a series of work to remove the GOVUK Frontend Toolkit library from our codebase:

https://www.pivotaltracker.com/story/show/182596710

We want to remove GOV.UK Frontend Toolkit from our code to replace it with GOV.UK Frontend.

This replaces all the colour variables and functions from toolkit with ones from GOV.UK Frontend (or hardcoded values where necessary). This is stage one - we have legacy mode on (`$govuk-compatibility-govukfrontendtoolkit: true;`) so we have access to the [legacy colour palette](https://github.com/alphagov/govuk-frontend/blob/062946035cf9d105b6ce045f4866927605d42675/src/govuk/settings/_colours-palette.scss#L70). At this point, the colours look the same. When we turn legacy mode off (stage two) we will need to use the [modern colour palette](https://github.com/alphagov/govuk-frontend/blob/062946035cf9d105b6ce045f4866927605d42675/src/govuk/settings/_colours-palette.scss#L37) and this will change the appearance of the colours.

There are a couple of places where things _have_ changed in this PR:
- where we use `govuk-tint()` it affects the colour now
- We were [using the new focus colour early](https://github.com/alphagov/notifications-admin/blob/89c6384ea04f95ec896cdd7b00dc0e31a6f26f1f/app/assets/stylesheets/govuk-frontend/focus/settings.scss), but were still using the old colour in `app/assets/stylesheets/govuk-elements/forms/_form-multiple-choice.scss`. This updates `app/assets/stylesheets/govuk-elements/forms/_form-multiple-choice.scss` to be consistent.

https://www.pivotaltracker.com/story/show/182596710

**Merge after**
- https://github.com/alphagov/notifications-admin/pull/4359 has been merged

### CODE REVIEW

File: app/assets/stylesheets/globals.scss [hunk 1]
```diff
@@ -60,10 +60,13 @@ input[type="search"]::-webkit-search-decoration {
 
 // Global styles for form controls
 
+// This is the focus colour from GOV.UK Frontend ($govuk-focus-colour).
+// We can't use the colour variable from GOV.UK Frontend, since this file is
+// above GOV.UK Frontend in the cascade.
```
Comment (tombye) [inline app/assets/stylesheets/globals.scss:65]: We should add a 'TODO' comment here saying these styles can be removed once all radios and textareas are converted to use GOVUK Frontend. We could also delete line 68 as we have no `<select>`s in this codebase.

**Review by tombye — CHANGES_REQUESTED** [review]
This is an epic piece of work and a lot of thought has gone into the detail so 🥇 for doing it.

I've added a few comments but nothing minor. I also spotted some colours hard-coded as hex values that should be recorded somehow so whoever turns the compatibility flag off, and so turns the new colours on, knows what else to update.

The hard-coded colours are:
- 'blue' appears as its hex value in:
  - app/templates/views/broadcast/macros/area-map.html
  - app/templates/views/broadcast/partials/area-map-javascripts.html
  - app/assets/images/folder-blue-bold.svg
  - app/assets/javascripts/main.js
  - app/config.py
  - app/assets/error_pages/5xx.html
  - app/assets/images/product/02-reporting.svg
  - app/assets/stylesheets/views/product-page.scss (as the blue specified as $product-page-blue at the top)
- 'light-blue' appears as its hex value in:
  - app/templates/views/broadcast/macros/area-map.html
  - app/templates/views/broadcast/partials/area-map-javascripts.html
  - app/assets/images/folder-blue-bold-hover.svg
- 'red' appears as its hex value in:
  - app/templates/views/find-users/user-information.html
- 'grey-1' appears as its hex value in:
  - app/assets/images/cross-grey.svg
  - app/assets/stylesheets/components/tick-cross.scss
- 'grey-3' appears as its hex value in:
  - app/assets/images/folder-black-bold.svg
  - app/assets/images/folder-black.svg
  - app/assets/images/folder-blue-bold.svg
  - app/assets/images/folder-blue-bold-hover.svg

I'd recommend mentioning each of these in the commit for their colour, unless another method of recording them would work better.

'black' also appears as its hex value in:
  - app/templates/admin_template.html
  - app/templates/vendor/govuk-frontend/template.njk
  - app/assets/images/folder-black-bold.svg
  - app/assets/images/security-key.svg
  - app/assets/error_pages/5xx.html
  - app/assets/stylesheets/govuk-elements/_forms.scss

I mention it separately because it won't change so neither will the hex values but it should be recorded in the same way for consistency.

The following SVGs from the product page (signedout) use various existing colours:
- app/assets/images/product/01-templates.svg
- app/assets/images/product/02-reporting.svg

I put together [a spreadsheet of the colour changes](https://docs.google.com/spreadsheets/d/1ejNMnHcBafmI5g9I8m-ppyInOjWRtt1gMNNzzVEHkJY/edit?usp=sharing), to help visualise the changes we'll need to do when removing the flag.

File: app/assets/stylesheets/components/vendor/breadcrumbs.scss [hunk 2]
```diff
@@ -67,7 +67,7 @@
     a:hover,
     a:active,
     a:visited {
-      color: $black;
+      color: govuk-colour("black");
```
[the remark this answers is not in the document]
Comment (klssmith) [reply app/assets/stylesheets/components/vendor/breadcrumbs.scss]: This is now updated to `$govuk-text-colour`

File: app/assets/stylesheets/app.scss [hunk 3]
```diff
@@ -237,7 +237,7 @@ details .arrow {
     height: 19px;
     border-radius: 50%;
     animation: live-pulse 1.5s infinite;
-    //box-shadow: inset 0 0 0 2px govuk-colour("red"), inset 0 0 0 5px $white;
+    //box-shadow: inset 0 0 0 2px govuk-colour("red"), inset 0 0 0 5px govuk-colour("white");
```
[the remark this answers is not in the document]
Comment (klssmith) [reply app/assets/stylesheets/app.scss]: 👍🚮

File: app/assets/stylesheets/globals.scss (hunk 1 above)
Comment (klssmith) [reply app/assets/stylesheets/globals.scss:67]: Now done

File: app/assets/stylesheets/components/banner.scss [hunk 4]
```diff
@@ -3,14 +3,14 @@
 .banner-default {
 
   @include govuk-font(19, $weight: bold);
-  color: $button-colour;
+  color: govuk-colour("green");
```
[the remark this answers is not in the document]
Comment (klssmith) [reply app/assets/stylesheets/components/banner.scss:6]: When I tried to replace `govuk-colour("green")` with `$govuk-success-colour`, I got the error that `$govuk-success-colour` is an undefined variable - I can't really see why, because it does seem to exist in the version of GOV.UK Frontend that we are using, and I can use other variables like `$govuk-error-colour` in this file.

File: app/assets/stylesheets/components/banner.scss [hunk 5]
```diff
@@ -90,7 +90,7 @@
   &-with-service-name {
     margin-top: -10px;
     padding-top: 0;
-    box-shadow: 0 -1px 0 0 darken($govuk-blue, 10%);
+    box-shadow: 0 -1px 0 0 darken($govuk-brand-colour, 10%);
```
[the remark this answers is not in the document]
Comment (klssmith) [reply app/assets/stylesheets/components/banner.scss:93]: `darken` and `govuk-tint` do slightly different things, so `darken($govuk-brand-colour, 10%)` and `govuk-shade($govuk-brand-colour, 10%)` don't give the same colour.

From trial and error, `govuk-shade($govuk-brand-colour, 30%)` looks close - do you reckon I should go with that, or keep this the way it is?

File: app/assets/stylesheets/components/banner.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (tombye) [reply app/assets/stylesheets/components/banner.scss:6]: Looks like the success colour was added in version 3.10: https://github.com/alphagov/govuk-frontend/blob/v3.10.0/src/govuk/settings/_colours-applied.scss

…so we can’t use it for now. Maybe just add a comment for future us, when we're able to bump GOVUK Frontend.

[the remark this answers is not in the document]
Comment (klssmith) [reply app/assets/stylesheets/components/banner.scss:7]: I'm not sure why I thought it was in our version, but that would explain why it's not working! I've added some TODO comments to the file and mentioned it in the commit message.

File: app/assets/stylesheets/components/banner.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (tombye) [reply app/assets/stylesheets/components/banner.scss:95]: I'm not sure I had a good look at where this is used when I made my previous comment, and when we talked about it. The `banner-tour` is actually used for the main content area of the tour for broadcasts services. This `box-shadow` is used when there's some content above it on a white background (the main content area is on a blue background), so make the join between those areas more distinct.

In other words it's super-snowflake territory and not really a case where we're specifying a colour, just adding a subtle shadow to an area. That being the case, I think we can keep it using `darken`.

**Review by tombye — APPROVED** [review]
All the comments have now been addressed and we've agreed (outside of this pull request) that the information about the hex codes is already in my previous comment so doesn't need adding anywhere else.

👍🏻
